x/interchaintxs/keeper/keeper.go

```diff
@@ -92,3 +92,14 @@ func (k Keeper) ChargeFee(ctx sdk.Context, payer sdk.AccAddress, fee sdk.Coins)
 func (k Keeper) GetAuthority() string {
 	return k.authority
 }
+
+func (k Keeper) GetLastCodeIDBeforeUpgrade(ctx sdk.Context) (codeID uint64) {
```

Comment:
I don't like the naming. which upgrade? what if we have more than one upgrade with some code ID? why do we even mention some upgrade in module keeper's interface? let's name it something like `GetLastFreeRegisterICACodeID` (there could definitely be a better naming for it, just an example) — says for itself what kind of ID it returns and hints at the context/reason for that